shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm

```diff
@@ -37,26 +37,23 @@ @implementation FlutterObservatoryPublisher
 #include <net/if.h>
 
 #include "flutter/fml/logging.h"
-#include "flutter/fml/make_copyable.h"
 #include "flutter/fml/memory/weak_ptr.h"
 #include "flutter/fml/message_loop.h"
 #include "flutter/fml/platform/darwin/scoped_nsobject.h"
-#include "flutter/fml/task_runner.h"
 #include "flutter/runtime/dart_service_isolate.h"
 
 @protocol FlutterObservatoryPublisherDelegate
-- (instancetype)initWithOwner:(FlutterObservatoryPublisher*)owner;
-- (void)publishServiceProtocolPort:(NSString*)uri;
+- (void)publishServiceProtocolPort:(NSURL*)uri;
 - (void)stopService;
-
-@property(readonly) fml::scoped_nsobject<NSURL> url;
```

Comment:
Removed the delegate `url` property and moved it up into `FlutterObservatoryPublisher`.  The delegates don't need to hang onto it.